### PR TITLE
More gui work

### PR DESCRIPTION
**Pull request by timhot**
Mostly finished work on new GUI so want to merge back into Master
Next - once back in Master, I'll add automatic testing back in and prune dead code/comments

### CODE REVIEW

File: app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                    } else if (messageType.equalsIgnoreCase("UPLOADING_FAILED_NOT_REGISTERED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("RECORDING_STARTED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("RECORDING_FINISHED")) {
+                        tvMessages.setText(messageToDisplay);
+                        //getView().findViewById(R.id.btnRecordNow).setEnabled(true);
+                        btnRecordNow.setEnabled(true);
+                        btnRecordNow.setVisibility(View.VISIBLE);
+                    }
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java:183]: if messageType is made uppercase first then you could use a switch statement here instead. That would probably be a little less verbose.

File: app/src/main/java/nz/org/cacophony/cacophonometer/TestingFragment.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+                if (!isChecked){
+                    Prefs prefs = new Prefs(getActivity().getApplicationContext());
+                    prefs.setUseShortRecordings(false);
+//                    swShortRecordings.setChecked(false);
+//                    swShortRecordings.setEnabled(false);
+
+                    prefs.setUseVeryFrequentRecordings(false);
+//                    swUseVeryFrequentRecordings.setChecked(false);
+//                    swUseVeryFrequentRecordings.setEnabled(false);
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/cacophonometer/TestingFragment.java:68]: Do these commented out sections have to stay around?

File: app/src/main/java/nz/org/cacophony/cacophonometer/Util.java [hunk 3]
```diff
@@ -251,15 +265,15 @@ static String getDeviceID(String webToken) throws Exception {
...
-    }
+//    static String getDeviceName(String webToken) throws Exception {
+//        if (webToken == null){
+//            return "";
+//        }
+//
+//        String webTokenBody = Util.decoded(webToken);
+//        JSONObject jObject = new JSONObject(webTokenBody);
+//        return jObject.getString("deviceName");
+//    }
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/cacophonometer/Util.java:276]: delete?

**Comment by timhot** [comment]
thanks.

Should I tidy those things up before the merge or after?

Tim

EDIT - Decided to:
1) Merge
2) Create a new branch where I am tiding up code and adding automated testing back in (old tests no longer work due to major GUI changes).

________________________________

File: app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java (hunk 1 above)
Comment (timhot) [reply app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java:183]: The current version of Java that Android uses does not allow switch with Strings.  I probably should have used enumerations... but won't change now.
https://stackoverflow.com/questions/14367629/android-coding-with-switch-string

File: app/src/main/java/nz/org/cacophony/cacophonometer/TestingFragment.java (hunk 2 above)
Comment (timhot) [reply app/src/main/java/nz/org/cacophony/cacophonometer/TestingFragment.java:68]: Removed them and have been through all other code and removed similar

File: app/src/main/java/nz/org/cacophony/cacophonometer/Util.java (hunk 3 above)
Comment (timhot) [reply app/src/main/java/nz/org/cacophony/cacophonometer/Util.java:276]: removed

File: app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java (hunk 1 above)
Comment (mjs) [reply app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java:183]: All good. This shows my relative inexperience with Java :)

**Pull request merged**